src/Compilers/Server/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -338,6 +339,7 @@ private void ChangeKeepAlive(TimeSpan? value)
         /// will never fail.  It will always produce a <see cref="ConnectionData"/> value.  Connection errors
         /// will end up being represented as <see cref="CompletionReason.ClientDisconnect"/>
         /// </summary>
+        [SuppressMessage("Usage", "VSTHRD003:Avoid awaiting foreign Tasks", Justification = "Needs review: https://github.com/dotnet/roslyn/issues/34287")]
```

Comment:
This method is only called with `Task` created in this file so it's not relevant to this warning. Feel like this warning is pretty broad.

Reply:
> Feel like this warning is pretty broad.

Agree. I have #34287 tracking items to review with @AArnott. Overall there were relatively few cases flagged by this analyzer considering the size of Roslyn.

Reply:
I get the critical piece here is thread affinity not where the task was created. In this program though, and for the majority of tasks in Roslyn, thread affinity is the uncommon case. The default is non-affinitized. 


---
In reply to: [268022749](https://github.com/dotnet/roslyn/pull/34342#discussion_r268022749) [](ancestors = 268022749)